src/main/java/jpos/config/simple/xml/JavaxRegPopulator.java

```diff
@@ -0,0 +1,513 @@
+        transformer.setOutputProperty( OutputKeys.DOCTYPE_SYSTEM, document.getDoctype().getSystemId() );
+        transformer.transform( source, new StreamResult( outputStream ) );
+    }
+
+    private Document CreateEmptyDocument()
+            throws ParserConfigurationException
+    {
+        DOMImplementation domImpl = DocumentBuilderFactory.newInstance().newDocumentBuilder().getDOMImplementation();
+        DocumentType docType = domImpl.createDocumentType( "JposEntries", DTD_DOC_TYPE_VALUE, DTD_FILE_NAME );
+
+        return domImpl.createDocument(null, "JposEntries", docType);
+    }
+
+
+    private void insertJposEntriesInDoc( Enumeration entries, Document doc )
+    {
+        while( entries.hasMoreElements() )
+        {
+            JposEntry jposEntry = (JposEntry) entries.nextElement();
+
+            if( JposEntryUtility.isValidJposEntry( jposEntry ) )
+            {
+                Element jposEntryElement = doc.createElement( "JposEntry" );
+
+                jposEntryElement.setAttribute( "logicalName", jposEntry.getLogicalName() );
+                insertJposPropertiesInElement( doc, jposEntry, jposEntryElement );
+                doc.getDocumentElement().appendChild( jposEntryElement );
+            }
+        }
+    }
+
+    private void insertJposPropertiesInElement( Document doc, JposEntry jposEntry, Element jposEntryElement )
+    {
+        Element creation = doc.createElement( "creation" );
+        Element jpos = doc.createElement( "jpos" );
+        Element product = doc.createElement( "product" );
+        Element vendor = doc.createElement( "vendor" );
+
+        for( Element tag : new Element[] { creation, vendor, jpos, product } )
+            jposEntryElement.appendChild(tag);
+
+        List<JposEntry.Prop> sortedProps = getSortedList( jposEntry.getProps() );
+        for( JposEntry.Prop prop : sortedProps )
+        {
+            if( notAttribute( prop, creation, JposEntry.SERVICE_CLASS_PROP_NAME, "serviceClass" ) &&
+                    notAttribute( prop, creation, JposEntry.SI_FACTORY_CLASS_PROP_NAME, "factoryClass" ) &&
+                    notAttribute( prop, vendor, JposEntry.VENDOR_NAME_PROP_NAME, "name" ) &&
+                    notAttribute( prop, vendor, JposEntry.VENDOR_URL_PROP_NAME, "url" ) &&
+                    notAttribute( prop, jpos, JposEntry.JPOS_VERSION_PROP_NAME, "version" ) &&
+                    notAttribute( prop, jpos, JposEntry.DEVICE_CATEGORY_PROP_NAME, "category" ) &&
+                    notAttribute( prop, product, JposEntry.PRODUCT_NAME_PROP_NAME, "name" ) &&
+                    notAttribute( prop, product, JposEntry.PRODUCT_DESCRIPTION_PROP_NAME, "description" ) &&
+                    notAttribute( prop, product, JposEntry.PRODUCT_URL_PROP_NAME, "url" )
+            )
+                addPropElement( doc, jposEntryElement, prop );
+        }
+    }
+
+    private List<JposEntry.Prop> getSortedList( Iterator props )
```

Comment:
Sorry I did not know Collection.sort, therefore I wrote my own sort. 
For me, it looks a little bit different: 

    private List<JposEntry.Prop> getSortedList(Iterator<JposEntry.Prop> props) {
        ArrayList<JposEntry.Prop> sortedProps = new ArrayList<>();
        props.forEachRemaining(sortedProps::add);
        Collections.sort(sortedProps, new Comparator<JposEntry.Prop>() {
            @Override
            public int compare(JposEntry.Prop o1, JposEntry.Prop o2) {
                return o1.getName().compareToIgnoreCase(o2.getName());
            }
        });
        return sortedProps;
    }
Is it what you mean?

Reply:
Your code is the same as mine but with an explicit implementation of the Single-method-Interface.
For such single-method-interface implementations since Java8 the lambda notation may be used. Both is right. Lambda is almost preferred, however.
You may leave this to me to change to lambda notation during merging if you are not familiar with lambda...